addons/arsenal/functions/fnc_updateUniqueItemsList.sqf

```diff
@@ -13,106 +14,103 @@
+            {!(_x in ((GVAR(virtualItems) select IDX_VIRT_ATTACHMENTS) select IDX_VIRT_MUZZLE_ATTACHMENTS))} &&
+            {_x in ((_configItems select IDX_VIRT_ATTACHMENTS) select IDX_VIRT_MUZZLE_ATTACHMENTS)}
+        ): {
+            ((GVAR(virtualItems) select IDX_VIRT_UNIQUE_ATTACHMENTS) select IDX_VIRT_MUZZLE_ATTACHMENTS) pushBackUnique _x;
         };
-        // acc
+        // Bipods
         case (
-                isClass (_configCfgWeapons >> _x) &&
-                {!(_x in ((GVAR(virtualItems) select IDX_VIRT_ATTACHEMENTS) select 3))} &&
-                {_x in ((_itemsCache select 1) select 3)}
-            ): {
-            ((GVAR(virtualItems) select 22) select 3) pushBackUnique _x;
+            _isWeapon &&
+            {!(_x in ((GVAR(virtualItems) select IDX_VIRT_ATTACHMENTS) select IDX_VIRT_BIPOD_ATTACHMENTS))} &&
+            {_x in ((_configItems select IDX_VIRT_ATTACHMENTS) select IDX_VIRT_BIPOD_ATTACHMENTS)}
+        ): {
+            ((GVAR(virtualItems) select IDX_VIRT_UNIQUE_ATTACHMENTS) select IDX_VIRT_BIPOD_ATTACHMENTS) pushBackUnique _x;
         };
-
-        // Misc
+        // Misc. items
         case (
-                isClass (_configCfgWeapons >> _x) &&
-                {!(_x in (GVAR(virtualItems) select 17))} &&
-                {!(_x in ((_itemsCache select 1) select 0))} &&
-                {!(_x in ((_itemsCache select 1) select 1))} &&
-                {!(_x in ((_itemsCache select 1) select 2))} &&
-                {!(_x in ((_itemsCache select 1) select 3))}
-            ): {
-            (GVAR(virtualItems) select 18) pushBackUnique _x;
+            _isWeapon &&
+            {!(_x in (GVAR(virtualItems) select IDX_VIRT_MISC_ITEMS))} &&
+            {!(_x in ((_configItems select IDX_VIRT_ATTACHMENTS) select IDX_VIRT_OPTICS_ATTACHMENTS))} &&
+            {!(_x in ((_configItems select IDX_VIRT_ATTACHMENTS) select IDX_VIRT_FLASHLIGHT_ATTACHMENTS))} &&
+            {!(_x in ((_configItems select IDX_VIRT_ATTACHMENTS) select IDX_VIRT_MUZZLE_ATTACHMENTS))} &&
+            {!(_x in ((_configItems select IDX_VIRT_ATTACHMENTS) select IDX_VIRT_BIPOD_ATTACHMENTS))}
+        ): {
+            (GVAR(virtualItems) select IDX_VIRT_UNIQUE_MISC_ITEMS) pushBackUnique _x;
         };
-
         // Backpacks
-        case (getNumber (_configVehicles >> _x >> "isBackpack") == 1): {
-            (GVAR(virtualItems) select 23) pushBackUnique _x;
+        case (getNumber (_cfgVehicles >> _x >> "isBackpack") == 1): {
+            (GVAR(virtualItems) select IDX_VIRT_UNIQUE_BACKPACKS) pushBackUnique _x;
         };
-
         // Facewear
-        case (isClass (_configGlasses >> _x)): {
-            (GVAR(virtualItems) select 24) pushBackUnique _x;
+        case (isClass (_cfgGlasses >> _x)): {
+            (GVAR(virtualItems) select IDX_VIRT_UNIQUE_GOGGLES) pushBackUnique _x;
         };
     };
-} foreach _array;
+} forEach (uniformItems GVAR(center) + vestItems GVAR(center) + backpackItems GVAR(center));
```

Comment:
Imagine being a medic, backpackItems array now has hundreds of bandages and medical supplies.
99% of them being duplicates

You will forEach through an array that likely has 90+% of duplicates where you only need one.

uniqueUnitItems command/cba thing does exist, if really getting items from the center unit is required here.
(Also can't center be a box? a box wouldn't have uniform/vest/backpack?)

Reply:
> Imagine being a medic, backpackItems array now has hundreds of bandages and medical supplies. 99% of them being duplicates
> 
> You will forEach through an array that likely has 90+% of duplicates where you only need one.
> 
> uniqueUnitItems command/cba thing does exist, if really getting items from the center unit is required here. (Also can't center be a box? a box wouldn't have uniform/vest/backpack?)

I must admit, I didn't think much about this.
I just "de-macrod" what was already there (as the macro was no longer needed) - in in turn means that I don't think it's possible for the center to be a box.

I fully agree and will check if a uniqueItems command/function is available for this purpose.